client/src/worker_registry/mod.rs

```diff
@@ -131,26 +154,65 @@ impl SlotManager {
         namespace: String,
         task_queue: String,
     ) -> Option<Box<dyn Slot + Send>> {
-        self.manager
+        self.slot_manager
             .read()
             .try_reserve_wft_slot(namespace, task_queue)
     }
 
     /// Register a local worker that can provide WFT processing slots.
-    pub fn register(&self, provider: Box<dyn SlotProvider + Send + Sync>) -> Option<WorkerKey> {
-        self.manager.write().register(provider)
+    pub fn register_slot(
```

Comment:
IMO since every worker registers for slots and for heartbeats, it'd be simpler to just have a single worker register and a single worker trait/struct that handles slots and worker heartbeats instead of splitting them.